src/i_video.c

```diff
@@ -725,6 +725,32 @@ static void I_ResetTargetRefresh(void);
  #define I_CpuPause()
 #endif
 
+NOINLINE static void I_WaitUntil(uint64_t target_time)
```

Comment:
Why do we need to disable `inline`? How can disabling `inline` for a single function affect performance?

Reply:
Moving the function out of line aligns it to a cache boundary so that the whole loop fits in a single cache line. I_FinishUpdate also benefits by moving the cold-path instructions out of line in the case where the limiter isn't being used.

Reply:
With the `/Gy-` flag this is less important because `noinline` doesn't change the layout of the function (it did before, despite not actually being inlined). But it still seems a good idea to keep it for future-proofing against future compiler changes, for the reasons mentioned above.

Reply:
> Moving the function out of line aligns it to a cache boundary so that the whole loop fits in a single cache line. I_FinishUpdate also benefits by moving the cold-path instructions out of line in the case where the limiter isn't being used.

`I_FinishUpdate` is only called once per frame, I don't think it would affect performance at all if all local variables were cached. We have functions that are called to draw a few pixels, in the case of these functions optimisations like this can make a difference.

The calculations in the inner loop are very minimal, the OS behaviour in `I_Sleep` affects the performance much more.

Reply:
I was surprised as well but it did make a difference in my testing before adding `/Gy-`. Performance is much less sensitive for me after the compiler flag fix so I don't particularly care about this `noinline` directive that much anymore, I was just giving you my rationale for keeping it.

Reply:
I have an unusually old CPU from like 2007 (i5-750 @ 3.6 GHz). But I still think Doom should run blazingly fast on such things.